lib/client/bst-proxy.ts

```diff
@@ -128,16 +129,17 @@ export class BSTProxy {
         return this;
     }
 
-    public start(onStarted?: (error?: any) => void): BSTProxy {
+    public async start(onStarted?: (error?: any) => void): Promise<BSTProxy> {
```

Comment:
We should not have the promise and the onstarted. Since most of these routines use callbacks, I think we should stick with that (for now).